internal/eval/eval.go

```diff
@@ -328,6 +330,153 @@ func evalGETDEL(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(obj.Value, false)
 }
 
+// evalJSONDEBUG reports value's memmory usage in bytes
+// Returns arity error if subcommand is missing
+// Supports only two subcommand as of now - HELP and MEMORY
+func evalJSONDebug(args []string, store *dstore.Store) []byte {
+	if len(args) < 1 {
+		return diceerrors.NewErrArity("JSON.DEBUG")
+	}
+	subcommand := strings.ToUpper(args[0])
+	switch subcommand {
+	case "HELP":
+		return evalJSONDebugHelp()
+	case "MEMORY":
+		return evalJSONDebugMemory(args[1:], store)
+	default:
+		return diceerrors.NewErrWithFormattedMessage("unknown subcommand - try `JSON.DEBUG HELP`")
+	}
+}
+
+// evalJSONDebugHelp implements HELP subcommand for evalJSONDebug
+// It returns help text
+// It ignore any other args
+func evalJSONDebugHelp() []byte {
+	memoryText := "MEMORY <key> [path] - reports memory usage"
+	helpText := "HELP                - this message"
+	message := []string{memoryText, helpText}
+	return clientio.Encode(message, false)
+}
+
+// evalJSONDebugMemory implements MEMORY subcommand for evalJSONDebug
+// It returns value's memory usage in bytes
+func evalJSONDebugMemory(args []string, store *dstore.Store) []byte {
+	if len(args) < 1 {
+		return diceerrors.NewErrArity("JSON.DEBUG")
+	}
+	key := args[0]
+
+	// default path is root if not specified
+	path := defaultRootPath
+	if len(args) > 1 {
+		path = args[1] // any more args are ignored for this command altogether
+	}
+
+	obj := store.Get(key)
+	if obj == nil {
+		return clientio.RespZero
+	}
+
+	// check if the object is a valid JSON
+	errWithMessage := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeJSON, object.ObjEncodingJSON)
+	if errWithMessage != nil {
+		return errWithMessage
+	}
+
+	// handle root path
+	if path == defaultRootPath {
+		jsonData := obj.Value
+
+		// memory used by json data
+		size := calculateMemory(jsonData)
+		if size == -1 {
+			return diceerrors.NewErrWithMessage("unknown type")
+		}
+
+		// add memory used by storage object
+		size += int(unsafe.Sizeof(obj)) + calculateMemory(obj.LastAccessedAt) + calculateMemory(obj.TypeEncoding)
+
+		return clientio.Encode(size, false)
+	}
+
+	// handle nested paths
+	var results []any = []any{}
+	if path != defaultRootPath {
+
+		// check if path is valid
+		expr, err := jp.ParseString(path)
+		if err != nil {
+			return diceerrors.NewErrWithMessage("invalid JSON path")
+		}
+
+		results = expr.Get(obj.Value)
+
+		// handle error cases
+		if len(results) == 0 {
+			// handle out of bound index path for array json type
+			isArray := utils.IsArray(obj.Value)
+			if isArray {
+				arr := obj.Value.([]any)
```

Comment:
You are casting types twice, once inside the `IsArray` method and once here. This adds to the latency.

You can replace the call to `IsArray` with the following:
```go
arr, ok := obj.Value.([]any)
if !ok {
  // return err
}
// use arr
```

Reply:
Done.